### PR TITLE
Add AccelerationStructurePassDescriptor

### PR DESCRIPTION
**Pull request by kvark**
Also renames "accelerator_structure" module for consistency.
Tested manually in https://github.com/kvark/blade/pull/179

### CODE REVIEW

**Comment by kvark** [comment]
@grovesNL could you take a look, please?

**Comment by kvark** [comment]
@grovesNL pinging again

File: src/acceleration_structure_pass.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+impl AccelerationStructurePassSampleBufferAttachmentDescriptor {
+    pub fn new() -> Self {
+        let class = class!(MTLAccelerationStructurePassSampleBufferAttachmentDescriptor);
+        unsafe { msg_send![class, new] }
+    }
+}
+
+impl AccelerationStructurePassSampleBufferAttachmentDescriptorRef {
+    pub fn sample_buffer(&self) -> &CounterSampleBufferRef {
```
Comment (madsmtm) [inline src/acceleration_structure_pass.rs:75]: This is nullable
```suggestion
    pub fn sample_buffer(&self) -> Option<&CounterSampleBufferRef> {
```

**Review by madsmtm — APPROVED** [review]

File: src/acceleration_structure_pass.rs (hunk 1 above)
Comment (kvark) [reply src/acceleration_structure_pass.rs:75]: Yes indeed, but that also applies to the other `sample_buffer` calls, so it's consistent.
I can switch them all to Options

**Comment by kvark** [comment]
Thanks for anybody looking at this. I'd be comfortable waiting more, but I want to get Zed up to date, and it would better for it to link to this repo's metal instead of my fork. So merging this. Please shout out if you feel this to be wrong!

**Pull request merged**